mutation_compactor.hh

```diff
@@ -150,10 +150,11 @@ template<emit_only_live_rows OnlyLive, compact_for_sstables SSTableCompaction>
 class compact_mutation_state {
     const schema& _schema;
     gc_clock::time_point _query_time;
-    gc_clock::time_point _gc_before;
     std::function<api::timestamp_type(const dht::decorated_key&)> _get_max_purgeable;
+    std::function<gc_clock::time_point(const dht::decorated_key&)> _get_repair_based_tombstone_gc_before;
```

Comment:
Let's just make _gc_before a function, mutation_compactor doesn't need to know about repair.

Reply:
It is already a function which returns a timestamp for a given key. But I can rename the variables to make repair disappear.